diff --git a/src/common/time_state.f90 b/src/common/time_state.f90
index 01ee63b87e3..e52b462abeb 100644
--- a/src/common/time_state.f90
+++ b/src/common/time_state.f90
@@ -162,8 +162,11 @@ contains
   !> Check if the simulation is done
   pure function time_state_is_done(this) result(is_done)
     class(time_state_t), intent(in) :: this
+    character(len=100) :: s
     logical :: is_done
 
+    write(s,*) this%t
+
     is_done = this%t - this%start_time .ge. this%end_time - this%start_time &
          .and. this%tstep .gt. 0
 
