diff --git a/src/common/time_state.f90 b/src/common/time_state.f90
index e11edc8ea3f..e04d1a0621b 100644
--- a/src/common/time_state.f90
+++ b/src/common/time_state.f90
@@ -155,9 +155,12 @@ contains
   !! the end time by almost `dt` and produce an output past it.
   pure function time_state_is_done(this) result(is_done)
     class(time_state_t), intent(in) :: this
+    character(len=100) :: s
     logical :: is_done
     real(kind=dp) :: span, tol
 
+    write(s,*) this%t
+
     span = abs(this%end_time - this%start_time)
     tol = 1.0e-9_dp * max(span, abs(this%dt))
 
