diff --git a/src/common/json_utils.f90 b/src/common/json_utils.f90
index 09b4a2dca19..3de6a7e2c43 100644
--- a/src/common/json_utils.f90
+++ b/src/common/json_utils.f90
@@ -267,7 +267,7 @@ contains
     end if
 
     call output%initialize()
-    call output%add(child)
+!    call output%add(child)
     nullify(child)
 
   end subroutine json_get_subdict
diff --git a/src/simulation_components/probes.F90 b/src/simulation_components/probes.F90
index f33542e5cb9..1e5470175c8 100644
--- a/src/simulation_components/probes.F90
+++ b/src/simulation_components/probes.F90
@@ -191,7 +191,7 @@ contains
        case ('point_zone')
           call this%read_point_zone(json_point, case%fluid%dm_Xh)
        case ('none')
-          call json_point%print()
+!          call json_point%print()
           call neko_error('No point type specified.')
        case default
           call neko_error('Unknown region type ' // point_type)
diff --git a/src/source_terms/brinkman_source_term.f90 b/src/source_terms/brinkman_source_term.f90
index c921c90de77..a2b409a39bd 100644
--- a/src/source_terms/brinkman_source_term.f90
+++ b/src/source_terms/brinkman_source_term.f90
@@ -167,7 +167,7 @@ contains
           call this%init_point_zone(object_settings)
 
          case ('none')
-          call object_settings%print()
+!          call object_settings%print()
           call neko_error('Brinkman source term objects require a region type')
          case default
           call neko_error('Brinkman source term unknown region type')
