diff --git a/src/fluid/bcknd/cpu/pnpn_res_cpu.f90 b/src/fluid/bcknd/cpu/pnpn_res_cpu.f90
index 7937285606..ad338a081b 100644
--- a/src/fluid/bcknd/cpu/pnpn_res_cpu.f90
+++ b/src/fluid/bcknd/cpu/pnpn_res_cpu.f90
@@ -1,7 +1,7 @@
 !> Residuals in the Pn-Pn formulation (CPU version)
 module pnpn_res_cpu
   use gather_scatter, only : gs_t, GS_OP_ADD
-  use operators, only : opgrad, curl, cdtp
+  use operators !, only : opgrad, curl, cdtp
   use field, only : field_t
   use ax_product, only : ax_t
   use coefs, only : coef_t
