diff --git a/src/common/signal.f90 b/src/common/signal.f90
index 0793c48a0..1419e9378 100644
--- a/src/common/signal.f90
+++ b/src/common/signal.f90
@@ -99,11 +99,11 @@ contains
 
     usr12 = sighdl_usr()
     
-    if (bge(usr12, usr)) then
-       raised = .true.
-    else
-       raised = .false.
-    end if
+  !  if (bge(usr12, usr)) then
+  !     raised = .true.
+  !  else
+  !     raised = .false.
+  !  end if
     
   end function signal_usr
   
